parachain/pallets/bridge/src/lib.rs

```diff
@@ -83,12 +82,12 @@ impl<T: Trait> Module<T> {
 		T::Verifier::verify(sender, app_id, &message)
 	}
 
-	fn dispatch(app: App, message: Message) -> DispatchResult {
-		match app {
-			App::ETH => T::AppETH::handle(message.payload),
-			App::ERC20 => T::AppERC20::handle(message.payload)
+	fn dispatch(address: &H160, message: &Message) -> DispatchResult {
+		match address {
+			_ if *address == T::AppETH::address() => T::AppETH::handle(message.payload.as_ref()),
+			_ if *address == T::AppERC20::address() => T::AppERC20::handle(message.payload.as_ref()),
+			_ => Err(Error::<T>::AppNotFound.into())
```

Comment:
Is there a reason you're using match instead of an if-else statement?

Reply:
Was mostly for aesthetic reasons. I like the match because it lines up the conditionals/statements more neatly than if-else. But I've reverted to if-else now as I admit in this case the code looks a bit convoluted.